src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -21,41 +23,61 @@ internal static IResourceBuilder<SqlServerServerResource> PublishAsAzureSqlDatab
 
         var configureConstruct = (ResourceModuleConstruct construct) =>
         {
-            var sqlServer = new SqlServer(construct, builder.Resource.Name);
-
-            sqlServer.AssignProperty(x => x.Administrators.AdministratorType, "'ActiveDirectory'");
-            sqlServer.AssignProperty(x => x.Administrators.IsAzureADOnlyAuthenticationEnabled, "true");
-            sqlServer.AssignProperty(x => x.Administrators.Sid, construct.PrincipalIdParameter);
-            sqlServer.AssignProperty(x => x.Administrators.Login, construct.PrincipalNameParameter);
-            sqlServer.AssignProperty(x => x.Administrators.TenantId, "subscription().tenantId");
-            sqlServer.AssignProperty(x => x.MinimalTlsVersion, "'1.2'");
-
-            sqlServer.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
-
-            var azureServicesFirewallRule = new SqlFirewallRule(construct, sqlServer, "AllowAllAzureIps");
-            azureServicesFirewallRule.AssignProperty(x => x.StartIPAddress, "'0.0.0.0'");
-            azureServicesFirewallRule.AssignProperty(x => x.EndIPAddress, "'0.0.0.0'");
+            var sqlServer = new SqlServer(builder.Resource.Name, AzureResourceVersions.SqlServerResourceVersion)
+            {
+                Administrators = new ServerExternalAdministrator()
+                {
+                    AdministratorType = SqlAdministratorType.ActiveDirectory,
+                    IsAzureADOnlyAuthenticationEnabled = true,
+                    Sid = construct.PrincipalIdParameter,
+                    Login = construct.PrincipalNameParameter,
+                    TenantId = BicepFunction.GetSubscription().TenantId
+                },
+                Version = "12.0",
+                PublicNetworkAccess = ServerNetworkAccessFlag.Enabled,
+                MinTlsVersion = SqlMinimalTlsVersion.Tls1_2,
+                Tags = { { "aspire-resource-name", construct.Resource.Name } }
+            };
+            construct.Add(sqlServer);
+
+            construct.Add(new SqlFirewallRule("sqlFirewallRule_AllowAllAzureIps", AzureResourceVersions.SqlFirewallRuleResourceVersion)
+            {
+                Parent = sqlServer,
+                Name = "AllowAllAzureIps",
+                StartIPAddress = new StringLiteral("0.0.0.0"),
+                EndIPAddress = new StringLiteral("0.0.0.0")
```

Comment:
@tg-msft is there a benefit to StringLiteral? Should there be an implicit conversion from strin?

Reply:
There is an implicit conversion from `T` - you should just be able to say `StartIPAddress = "0.0.0.0"`.

Reply:
For Sql, I can change this to the simple `StartIPAddress = "0.0.0.0"`. But that doesn't work for PostgreSQL because Start/End IPAddress are typed as `public BicepValue<IPAddress> StartIPAddress`. 

![image](https://github.com/user-attachments/assets/543ac248-92f9-40cd-8ca0-1c1ade16b479)

Reply:
Set it to `new IPAddress([0, 0, 0, 0])` and it should turn into `'0.0.0.0'` in Bicep.

Reply:
That's better than `new StringLiteral("0.0.0.0")`?

Reply:
Also - why is it different between Sql and PostgreSQL? Shouldn't that be addressed in the CDK?

Reply:
Changed to use `new IPAddress([0, 0, 0, 0])` for PostgreSQL